pkg/cmd/pr/checks/checks.go

```diff
@@ -65,208 +78,77 @@ func NewCmdChecks(f *cmdutil.Factory, runF func(*ChecksOptions) error) *cobra.Co
-		return fmt.Errorf("no commit found on the pull request")
-	}
+	return opts.Browser.Browse(openURL)
+}
 
-	rollup := pr.StatusCheckRollup.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
-	if len(rollup) == 0 {
-		return fmt.Errorf("no checks reported on the '%s' branch", pr.BaseRefName)
+func checksRun(opts *ChecksOptions) error {
+	if opts.WebMode {
+		return checksRunWebMode(opts)
 	}
 
-	passing := 0
-	failing := 0
-	skipping := 0
-	pending := 0
-
-	type output struct {
-		mark      string
-		bucket    string
-		name      string
-		elapsed   string
-		link      string
-		markColor func(string) string
-	}
+	var checks []check
+	var counts checkCounts
 
-	cs := opts.IO.ColorScheme()
-
-	outputs := []output{}
-
-	checkContexts := pr.StatusCheckRollup.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
-	for _, c := range eliminateDuplicates(checkContexts) {
-		mark := "✓"
-		bucket := "pass"
-		state := c.State
-		markColor := cs.Green
-		if state == "" {
-			if c.Status == "COMPLETED" {
-				state = c.Conclusion
-			} else {
-				state = c.Status
-			}
+	for {
+		findOptions := shared.FindOptions{
+			Selector: opts.SelectorArg,
+			Fields:   []string{"number", "baseRefName", "statusCheckRollup"},
 		}
-		switch state {
-		case "SUCCESS":
-			passing++
-		case "SKIPPED", "NEUTRAL":
-			mark = "-"
-			markColor = cs.Gray
-			skipping++
-			bucket = "skipping"
-		case "ERROR", "FAILURE", "CANCELLED", "TIMED_OUT", "ACTION_REQUIRED":
-			mark = "X"
-			markColor = cs.Red
-			failing++
-			bucket = "fail"
-		default: // "EXPECTED", "REQUESTED", "WAITING", "QUEUED", "PENDING", "IN_PROGRESS", "STALE"
-			mark = "*"
-			markColor = cs.Yellow
-			pending++
-			bucket = "pending"
+		pr, _, err := opts.Finder.Find(findOptions)
+		if err != nil {
+			return err
 		}
 
-		elapsed := ""
-		zeroTime := time.Time{}
-
-		if c.StartedAt != zeroTime && c.CompletedAt != zeroTime {
-			e := c.CompletedAt.Sub(c.StartedAt)
-			if e > 0 {
-				elapsed = e.String()
-			}
+		checks, counts, err = aggregateChecks(pr)
+		if err != nil {
+			return err
 		}
 
-		link := c.DetailsURL
-		if link == "" {
-			link = c.TargetURL
+		printSummary(opts.IO, counts)
+		err = printTable(opts.IO, checks)
+		if err != nil {
+			return err
 		}
 
-		name := c.Name
-		if name == "" {
-			name = c.Context
+		if counts.Pending == 0 || !opts.Watch {
+			break
 		}
 
-		outputs = append(outputs, output{mark, bucket, name, elapsed, link, markColor})
-	}
+		time.Sleep(opts.Interval)
 
-	sort.Slice(outputs, func(i, j int) bool {
-		b0 := outputs[i].bucket
-		n0 := outputs[i].name
-		l0 := outputs[i].link
-		b1 := outputs[j].bucket
-		n1 := outputs[j].name
-		l1 := outputs[j].link
-
-		if b0 == b1 {
-			if n0 == n1 {
-				return l0 < l1
-			}
-			return n0 < n1
+		if err := opts.IO.EnableVirtualTerminalProcessing(); err == nil {
+			// ANSI escape code to clean terminal window
+			// https://stackoverflow.com/a/22892171
+			fmt.Print(opts.IO.Out, "\033[H\033[2J")
```

Comment:
@vilmibm Can I get a double check on if we should be using this code to clear the screen or the one you used for `run watch`.

Reply:
we should be copying what I did for run watch (including the platform negotiation).